### ISSUE TITLE
Core tools dbcontext scaffold - unable to find provider EntityFramework.MicrosoftSqlServer

### ISSUE THREAD

**Issue by mrmeek** [dotnet/efcore#4721]
When I exec

`dotnet ef dbcontext scaffold "Server=localhost;Database=Test;Integrated Security=True;" EntityFramework.MicrosoftSqlServer`

I get 

> System.AggregateException: One or more errors occurred. (Unable to find provider assembly with name EntityFramework.MicrosoftSqlServer. Ensure the specified name is correct and is referenced by the project.)

I believe the provider was renamed from EntityFramework.SqlServer to EntityFramework.MicrosoftSqlServer for RC1, was it renamed again (or moved) for RC2?

Details:
`dotnet --version`

> .NET Command Line Tools (1.0.0-beta-001661)
> 
> Product Information:
>  Version:     1.0.0-beta-001661
>  Commit Sha:  2b90a2f831
> 
> Runtime Environment:
>  OS Name:     Windows
>  OS Version:  6.1.7601
>  OS Platform: Windows
>  Runtime Id:  win7-x64

``` xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
   <packageSources>
    <clear />
    <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
    <add key="dotnet-cli" value="https://www.myget.org/F/dotnet-cli/api/v3/index.json" />
    <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" />
    <add key="aspnetcidev" value="https://www.myget.org/F/aspnetcidev/api/v3/index.json" />
   </packageSources>
</configuration>
```

``` json
{
    "dependencies": {
        "Microsoft.AspNetCore.Diagnostics": "1.0.0-rc2-*",
        "Microsoft.AspNetCore.Hosting": "1.0.0-rc2-*",
        "Microsoft.AspNetCore.IISPlatformHandler": "1.0.0-rc2-*",
        "Microsoft.AspNetCore.Mvc": "1.0.0-rc2-*",
        "Microsoft.AspNetCore.Server.Kestrel": "1.0.0-rc2-*",
        "Microsoft.Extensions.Logging.Console": "1.0.0-*",
        "Microsoft.Extensions.Logging.Debug": "1.0.0-*",
        "Microsoft.EntityFrameworkCore": "1.0.0-*",
        "Microsoft.EntityFrameworkCore.SqlServer": "1.0.0-*",
        "Microsoft.EntityFrameworkCore.SqlServer.Design": "1.0.0-*"
    },
    "tools": {
        "dotnet-ef":  "1.0.0-*"
    },
    "frameworks": {
        "netstandardapp1.5": {
            "imports": [
                "dnxcore50",
                "portable-net451+win8"
            ],
            "dependencies": {
                "NETStandard.Library": "1.0.0-*"
            }
        }
    }
}
```

**Comment by bricelam** [comment]
The name of the provider in RC2 is `Microsoft.EntityFrameworkCore.SqlServer`

**Comment by mrmeek** [comment]
Thanks Brice!

...and wouldn't it figure, I actually tried that name already after making the assumption that it would be the most natural change. I didn't realize the correct name was resulting in a different error (shame on me for looking over the red text too quickly):

`dotnet ef dbcontext scaffold "Server=localhost;Database=test;Integrated Security=True;" "Microsoft.EntityFrameworkCore.SqlServer"`

>System.AggregateException: One or more errors occurred. (Could not load file or assembly 'System.Data.SqlClient, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

It's interesting, there are actually references to both an RC2 and an RC3 version of SQlClient v4.0.* in project.lock... I don't yet know how dependency resolution works well enough to decide if it's actually pulling down and/or referencing the wrong version of the assembly.

This is probably related: #4659

**Comment by bricelam** [comment]
That error should be resolved by #4713. Although you may start hitting yet another error about `sni.dll`... :smirk: I'm still debugging through that one.

**Comment by bricelam** [comment]
This issue can probably be closed as a dupe of #3925 now.

**Comment by mrmeek** [comment]
I pulled from volatiledev and did end up getting blocked by the sni.dll issue. Is that issue being tracked? I'd like to keep an eye on it
